lib/ui/pages/chat_page.dart

```diff
@@ -30,49 +27,63 @@ class _ChatPageState extends State<ChatPage> {
 
   @override
   Widget build(BuildContext context) {
-    return Scaffold(
-      backgroundColor: AppColors.darkGrey800,
-      appBar: AppBar(
-        backgroundColor: AppColors.darkGrey800,
-        title: Text(
-          'AI Provider',
-          style: AppTextStyles.h1.copyWith(color: AppColors.white),
-        ),
-        centerTitle: true,
-        elevation: 2,
-        scrolledUnderElevation: 0,
-        surfaceTintColor: Colors.transparent,
-      ),
-      body: GestureDetector(
-        behavior: HitTestBehavior.opaque,
-        onTap: () => FocusScope.of(context).unfocus(),
-        child: Column(
-          children: [
-            Expanded(
-              child: _messages.isEmpty
-                  ? const Center(
-                      child: Text(
-                        'Ask something!',
-                        style: AppTextStyles.backgoundHint,
-                      ),
-                    )
-                  : ListView.builder(
-                      controller: _scrollController,
-                      padding: const EdgeInsets.all(16),
-                      itemCount: _messages.length,
-                      itemBuilder: (context, index) {
-                        final message = _messages[index];
-                        return _buildMessageBubble(message);
-                      },
-                    ),
+    return BlocProvider(
+      create: (_) => ChatCubit(),
+      child: BlocConsumer<ChatCubit, ChatState>(
+        listener: (context, state) {
+          _scrollToBottom();
```

Comment:
I dont think that it is a good idea. listener will trigger on every state change. if now it's good, we will add some other params in future updates at unexpected bihaviour will occured